py3dtilers/ObjTiler/obj.py

```diff
@@ -110,22 +114,31 @@ def retrieve_objs(files, with_texture=False):
         Create Obj instance from OBJ file(s).
         :param files: paths to files
         :param with_texture: a boolean indicating if the textures should be read
-
         :return: a list of Obj.
         """
         objects = list()
 
         for obj_file in files:
             print("Reading " + str(obj_file))
-            geom = pywavefront.Wavefront(obj_file, collect_faces=True)
+            geom = pywavefront.Wavefront(obj_file, collect_faces=True, create_materials=True)
             if len(geom.vertices) == 0:
                 continue
+            gltfMaterials = []
+            mesh_index = 1
+
             for mesh in geom.mesh_list:
-                # Get id from its name
-                id = mesh.name
-                obj = Obj(id)
-                # Create geometry as expected from GLTF from an obj file
-                if obj.parse_geom(mesh, with_texture):
-                    objects.append(obj)
-
-        return Objs(objects)
+                for mesh_mat in mesh.materials:
+                    # get id from its name
+                    id = mesh_mat.name
+                    obj = Obj(id)
+                    obj.set_material_index(mesh_index)
+                    mesh_index += 1
+                    if obj.parse_geom(mesh_mat, with_texture):
+                        objects.append(obj)
+                    material = GlTFMaterial(rgb=[mesh.diffuse[0], mesh.diffuse[1], mesh.diffuse[2]], alpha=1. - mesh.diffuse[3], metallicFactor=0.)
```

Comment:
One last change: in line 138, all occurences of `mesh` should be `mesh_mat`

Reply:
my bad. it should be fine now.